client/lib/purchases/index.ts

```diff
@@ -327,16 +344,16 @@ function isObject( value ) {
  * @param {object} purchase - the purchase with which we are concerned
  * @returns {boolean}  True if the provided purchase is monthly, or false if not
  */
-function isMonthlyPurchase( purchase ) {
+export function isMonthlyPurchase( purchase: Purchase ): boolean {
 	const plan = getPlan( purchase.productSlug );
-	if ( isObject( plan ) ) {
+	if ( plan ) {
 		return isMonthlyPlan( purchase.productSlug );
 	}
 
 	// Note that getProductFromSlug() returns a string when given a non-product
 	// slug, so we need to check that it's an object before using it.
 	const product = getProductFromSlug( purchase.productSlug );
-	if ( isObject( product ) ) {
+	if ( product && typeof product !== 'string' ) {
```

Comment:
`getProductFromSlug()` can return a string if it fails, so here we make sure that it is not a string, which in turn should guarantee that it is an object without the need for type juggling in `isObject()`.